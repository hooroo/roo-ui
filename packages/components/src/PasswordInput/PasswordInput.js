import React, { Component } from 'react';
import styled from 'styled-components';
import { themeGet } from 'styled-system';
import { rem } from 'polished';
import cleanElement from 'clean-element';

import { Input, Icon, NakedButton } from '..';

const CleanInput = cleanElement(Input);
CleanInput.propTypes = Input.propTypes;
CleanInput.displayName = 'PasswordInput__Input';

const Wrapper = styled.div`
  position: relative;
`;

const Toggle = styled(NakedButton)`
  position: absolute;
  height: 100%;
  top: 0;
  left: calc(100% - ${rem('80px')});
  font-size: ${themeGet('fontSizes.sm')};
  color: ${themeGet('colors.greys.steel')};
  text-decoration: underline;

  &:focus {
    color: ${themeGet('colors.greys.charcoal')};
    outline: none;
  }
`;

class Base extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
    };

    this.toggleVisibility = this.toggleVisibility.bind(this);
  }

  toggleVisibility() {
    this.setState({
      visible: !this.state.visible,
    });
  }

  render() {
    return (
      <Wrapper>
        <CleanInput {...this.props} type={this.state.visible ? 'text' : 'password'} />

        <Toggle onClick={this.toggleVisibility}>
          <Icon name={this.state.visible ? 'visibilityOff' : 'visibility'} size={22} mr={2} />

          {this.state.visible ? 'Hide' : 'Show'}
        </Toggle>
      </Wrapper>
    );
  }
}

const PasswordInput = Input.withComponent(Base);

PasswordInput.defaultProps = {
  pr: rem('90px'),
};

export default PasswordInput;
