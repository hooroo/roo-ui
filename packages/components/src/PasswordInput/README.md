# Password input

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { PasswordInput } from '@roo-ui/components';

export default (
  <PasswordInput placeholder="Hello world" />
);
```

## Properties

| Name    | Description        | Type   | Default | Required? |
|:--------|:-------------------|:-------|:--------|:----------|
| `error` | render error style | `bool` | false   | -         |

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [space](https://github.com/jxnblk/styled-system/blob/v2.2.5/README.md#space-responsive).
