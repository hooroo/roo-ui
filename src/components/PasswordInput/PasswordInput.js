import React, { Component } from 'react';
import styled from '@emotion/styled';
import { themeGet } from '@styled-system/theme-get';
import { rem } from 'polished';
import { Input, Icon, NakedButton } from '../';

const StyledInput = styled(Input)``;

StyledInput.propTypes = Input.propTypes;
StyledInput.displayName = 'PasswordInput__Input';

const Wrapper = styled.div`
  position: relative;
`;

const Toggle = styled(NakedButton)`
  position: absolute;
  height: 100%;
  top: 0;
  left: calc(100% - ${rem('80px')});
  font-size: ${themeGet('fontSizes.sm')};
  color: ${themeGet('colors.greys.steel')};
  text-decoration: underline;

  &:focus {
    color: ${themeGet('colors.greys.charcoal')};
    outline: none;
  }
`;
Toggle.displayName = 'PasswordInput__Toggle';

class Base extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
    };

    this.toggleVisibility = this.toggleVisibility.bind(this);
  }

  toggleVisibility() {
    this.setState({
      visible: !this.state.visible,
    });
  }

  render() {
    console.log(this.props);
    return (
      <Wrapper>
        <StyledInput
          {...this.props}
          type={this.state.visible ? 'text' : 'password'}
        />

        <Toggle onClick={this.toggleVisibility}>
          <Icon
            name={this.state.visible ? 'visibilityOff' : 'visibility'}
            size={22}
            mr={2}
          />

          {this.state.visible ? 'Hide' : 'Show'}
        </Toggle>
      </Wrapper>
    );
  }
}

const PasswordInput = Input.withComponent(Base);

PasswordInput.defaultProps = {
  pr: rem('90px'),
};

export default PasswordInput;
