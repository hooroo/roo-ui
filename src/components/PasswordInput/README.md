# Password input

<!-- STORY -->

## Installation

```shell
$ yarn add roo-ui
```

## Example

```js
import { PasswordInput } from 'roo-ui/components';

export default <PasswordInput placeholder="Hello world" />;
```

## Properties

| Name    | Description        | Type   | Default | Required? |
| :------ | :----------------- | :----- | :------ | :-------- |
| `error` | render error style | `bool` | false   | -         |

## Customization

This component can be customized with [styled-system](https://jxnblk.com/styled-system) by passing props for [space](https://jxnblk.com/styled-system#space-theming).
