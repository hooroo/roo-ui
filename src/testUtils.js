import React from 'react';
import { mount, shallow } from 'enzyme';

export const mountWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return mount(themedTree);
};

export const shallowWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return shallow(themedTree);
};
