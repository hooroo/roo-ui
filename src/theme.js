import { rem } from 'polished';
import range from 'lodash/range';

import * as icons from './defaultIcons';

const namedColors = {
  white: '#FFFFFF',
  brightSun: '#FBCB3B',
  red: '#E00E00',
  maroon: '#870000',
  blue: '#8DE2E0',
  lightBlue: '#E7F7F7',
  orange: '#ED710B',
  lightOrange: '#FCEBCD',
  green: '#35A509',
  lightGreen: '#DEF2DE',
};

const brand = {
  primary: '#E40000',
  secondary: '#8DE2E0',
  tertiary: '#D20000',
  quaternary: '#F9F3E9',
};

const greys = {
  charcoal: '#323232',
  steel: '#666666',
  alto: '#DADADA',
  porcelain: '#F4F5F6',
  dusty: '#999999',
  mineShaft: '#222222',
};

const ui = {
  error: namedColors.orange,
  errorBackground: namedColors.lightOrange,
  info: namedColors.blue,
  infoBackground: namedColors.lightBlue,
  link: namedColors.red,
  linkHover: namedColors.maroon,
  success: namedColors.green,
  successBackground: namedColors.lightGreen,
};

const colors = {
  brand,
  greys,
  ui,
  ...namedColors,
};

const alertStyles = {
  default: { bg: 'greys.porcelain' },
  error: {
    bg: 'ui.errorBackground',
    icon: {
      name: 'warning',
      color: 'ui.error',
    },
  },
  info: {
    bg: 'ui.infoBackground',
    icon: {
      name: 'info',
      color: 'greys.charcoal',
    },
  },
  success: {
    bg: 'ui.successBackground',
    icon: {
      name: 'checkCircle',
      color: 'ui.success',
    },
  },
};

const borders = [0, '1px solid', '2px solid', '3px solid'];

const breakpoints = ['40rem', '52rem', '64rem'];

const buttons = {
  default: {
    color: 'white',
    backgroundColor: 'greys.charcoal',
  },
  primary: {
    color: 'white',
    backgroundColor: 'brand.primary',
  },
};

const fontFamily = 'Ciutadella, sans-serif';

const fontSizes = {
  xs: rem('13px'),
  sm: rem('15px'),
  base: rem('16px'),
  lg: rem('18px'),
  xl: rem('22px'),
  '2xl': rem('26px'),
  '3xl': rem('36px'),
  '4xl': rem('46px'),
};

const fontWeights = {
  normal: 500,
  bold: 600,
};

const letterSpacings = {
  normal: '0',
  wide: '0.125em',
};

const lineHeights = {
  tight: 1,
  normal: 1.35,
  loose: 1.5,
};

const maxWidths = {
  default: rem('1280px'),
};

const mediaQueries = breakpoints.map(
  n => `@media screen and (min-width: ${n})`,
);

const opacity = {
  disabled: 0.7,
};

const radii = {
  none: 0,
  default: '4px',
  rounded: '1000px',
};

const shadows = {
  none: 0,
  default: '0 1px 1px 0 rgba(0, 0, 0, 0.1)',
  heavy: '0 2px 2px 0 rgba(0, 0, 0, 0.1)',
  focus: `0 0 2px 2px ${colors.greys.dusty};`,
};

const space = range(0, 129, 4).map(i => rem(`${i}px`));

const textStyles = {
  caps: {
    textTransform: 'uppercase',
    fontWeight: fontWeights.bold,
  },

  h1: {
    fontSize: fontSizes['3xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h2: {
    fontSize: fontSizes['2xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h3: {
    fontSize: fontSizes.xl,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h4: {
    fontSize: fontSizes.lg,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h5: {
    fontSize: fontSizes.base,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h6: {
    fontSize: fontSizes.sm,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  label: {
    fontSize: fontSizes.sm,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.bold,
    color: colors.greys.charcoal,
  },
  text: {
    fontSize: fontSizes.base,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.normal,
  },
};

const transitions = {
  default: '200ms ease-in',
};

export default {
  alertStyles,
  borders,
  breakpoints,
  buttons,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  icons,
  letterSpacings,
  lineHeights,
  maxWidths,
  mediaQueries,
  opacity,
  radii,
  shadows,
  space,
  textStyles,
  transitions,
};
