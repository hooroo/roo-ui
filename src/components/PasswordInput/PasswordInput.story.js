import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import PasswordInput from './';
import README from './README.md';

storiesOf('Components/PasswordInput', module)
  .addDecorator(withDocs(README))
  .add('default', () => <PasswordInput placeholder="Hello world" />);
