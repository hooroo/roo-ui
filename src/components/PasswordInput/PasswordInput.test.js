import React from 'react';
import theme from 'theme';
import { mountWithTheme } from 'testUtils';
import { axe } from 'jest-axe';

import PasswordInput from '.';

describe('<PasswordInput />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(<PasswordInput />, theme);
  });

  it('has no accessibility errors', async () => {
    wrapper = mountWithTheme(
      <label htmlFor="input">
        Label
        <PasswordInput id="input" />
      </label>,
      theme,
    );
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('when visibility is not toggled', () => {
    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('on visibility toggle click', () => {
      beforeEach(() => {
        wrapper.find('PasswordInput__Toggle').simulate('click');
        wrapper.update();
      });

      it('sets <input /> type to text', () => {
        expect(wrapper.find('PasswordInput__Input').props()).toHaveProperty(
          'type',
          'text',
        );
      });
    });
  });

  describe('when visibility is toggled', () => {
    beforeEach(() => {
      wrapper.find('PasswordInput__Toggle').simulate('click');
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('on visibility toggle click', () => {
      beforeEach(() => {
        wrapper.find('PasswordInput__Toggle').simulate('click');
      });

      it('sets <input /> type to password', () => {
        expect(
          wrapper
            .update()
            .find('PasswordInput__Input')
            .props(),
        ).toHaveProperty('type', 'password');
      });
    });
  });
});
