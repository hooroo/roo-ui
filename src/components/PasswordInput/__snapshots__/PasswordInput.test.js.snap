// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> when visibility is not toggled renders correctly 1`] = `
.emotion-3 {
  padding-right: 5.625rem;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color 200ms ease-in;
  transition: border-color 200ms ease-in;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-3:focus {
  border-color: #8DE2E0;
}

.emotion-3:disabled {
  cursor: not-allowed;
  color: #666666;
  background-color: #F4F5F6;
}

.emotion-3::-webkit-input-placeholder {
  color: #666666;
}

.emotion-3::-moz-placeholder {
  color: #666666;
}

.emotion-3:-ms-input-placeholder {
  color: #666666;
}

.emotion-3::placeholder {
  color: #666666;
}

.emotion-3::-ms-clear {
  display: none;
}

.emotion-16 {
  position: relative;
}

.emotion-1 {
  margin-bottom: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 32px;
  padding-right: 32px;
  background-color: white;
  color: greys.charcoal;
  font-size: base;
  line-height: normal;
  text-align: left;
  border: 2px;
  border-color: greys.alto;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color;
  transition: border-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 5.625rem;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color 200ms ease-in;
  transition: border-color 200ms ease-in;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1:focus {
  border-color: #8DE2E0;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: #666666;
  background-color: #F4F5F6;
}

.emotion-1::-webkit-input-placeholder {
  color: #666666;
}

.emotion-1::-moz-placeholder {
  color: #666666;
}

.emotion-1:-ms-input-placeholder {
  color: #666666;
}

.emotion-1::placeholder {
  color: #666666;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-14 {
  border: none;
  width: auto;
  overflow: visible;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  padding: 0;
  margin: 0;
  background-color: transparent;
  color: inherit;
  position: absolute;
  height: 100%;
  top: 0;
  left: calc(100% - 5rem);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-14:disabled {
  cursor: not-allowed;
}

.emotion-14:hover:disabled {
  background-color: transparent;
  color: inherit;
}

.emotion-14:focus {
  outline: none;
}

.emotion-8 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
}

.emotion-6 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
}

<PasswordInput
  pr="5.625rem"
>
  <Base
    className="emotion-3 emotion-0"
  >
    <Wrapper>
      <div
        className="emotion-16 emotion-17"
      >
        <PasswordInput__Input
          bg="white"
          border={2}
          borderColor="greys.alto"
          className="emotion-3 emotion-0"
          color="greys.charcoal"
          fontSize="base"
          lineHeight="normal"
          mb={3}
          px={4}
          py={3}
          textAlign="left"
          type="password"
        >
          <input
            className="emotion-0 emotion-1 emotion-2"
            type="password"
          />
        </PasswordInput__Input>
        <PasswordInput__Toggle
          as="button"
          bg="transparent"
          color="inherit"
          m={0}
          onClick={[Function]}
          p={0}
          type="button"
        >
          <button
            className="emotion-14 emotion-15"
            color="inherit"
            onClick={[Function]}
            type="button"
          >
            <Icon
              mr={2}
              name="visibility"
              size={22}
              title={null}
            >
              <WithTheme(Component)
                className="emotion-8 emotion-5"
                mr={2}
                name="visibility"
                size={22}
                title={null}
              >
                <Component
                  className="emotion-8 emotion-5"
                  mr={2}
                  name="visibility"
                  size={22}
                  title={null}
                >
                  <StyledSvg
                    className="emotion-8 emotion-5"
                    fill="currentcolor"
                    height={22}
                    mr={2}
                    title="visibility"
                    viewBox="0 0 24 24"
                    width={22}
                  >
                    <svg
                      className="emotion-5 emotion-6 emotion-7"
                      fill="currentcolor"
                      height={22}
                      title="visibility"
                      viewBox="0 0 24 24"
                      width={22}
                    >
                      <path
                        d={null}
                      />
                    </svg>
                  </StyledSvg>
                </Component>
              </WithTheme(Component)>
            </Icon>
            Show
          </button>
        </PasswordInput__Toggle>
      </div>
    </Wrapper>
  </Base>
</PasswordInput>
`;

exports[`<PasswordInput /> when visibility is toggled renders correctly 1`] = `
.emotion-3 {
  padding-right: 5.625rem;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color 200ms ease-in;
  transition: border-color 200ms ease-in;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-3:focus {
  border-color: #8DE2E0;
}

.emotion-3:disabled {
  cursor: not-allowed;
  color: #666666;
  background-color: #F4F5F6;
}

.emotion-3::-webkit-input-placeholder {
  color: #666666;
}

.emotion-3::-moz-placeholder {
  color: #666666;
}

.emotion-3:-ms-input-placeholder {
  color: #666666;
}

.emotion-3::placeholder {
  color: #666666;
}

.emotion-3::-ms-clear {
  display: none;
}

.emotion-16 {
  position: relative;
}

.emotion-1 {
  margin-bottom: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 32px;
  padding-right: 32px;
  background-color: white;
  color: greys.charcoal;
  font-size: base;
  line-height: normal;
  text-align: left;
  border: 2px;
  border-color: greys.alto;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color;
  transition: border-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 5.625rem;
  display: block;
  width: 100%;
  outline: 0;
  -webkit-transition: border-color 200ms ease-in;
  transition: border-color 200ms ease-in;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-1:focus {
  border-color: #8DE2E0;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: #666666;
  background-color: #F4F5F6;
}

.emotion-1::-webkit-input-placeholder {
  color: #666666;
}

.emotion-1::-moz-placeholder {
  color: #666666;
}

.emotion-1:-ms-input-placeholder {
  color: #666666;
}

.emotion-1::placeholder {
  color: #666666;
}

.emotion-1::-ms-clear {
  display: none;
}

.emotion-14 {
  border: none;
  width: auto;
  overflow: visible;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  padding: 0;
  margin: 0;
  background-color: transparent;
  color: inherit;
  position: absolute;
  height: 100%;
  top: 0;
  left: calc(100% - 5rem);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-14:disabled {
  cursor: not-allowed;
}

.emotion-14:hover:disabled {
  background-color: transparent;
  color: inherit;
}

.emotion-14:focus {
  outline: none;
}

.emotion-8 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
}

.emotion-6 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
}

<PasswordInput
  pr="5.625rem"
>
  <Base
    className="emotion-3 emotion-0"
  >
    <Wrapper>
      <div
        className="emotion-16 emotion-17"
      >
        <PasswordInput__Input
          bg="white"
          border={2}
          borderColor="greys.alto"
          className="emotion-3 emotion-0"
          color="greys.charcoal"
          fontSize="base"
          lineHeight="normal"
          mb={3}
          px={4}
          py={3}
          textAlign="left"
          type="text"
        >
          <input
            className="emotion-0 emotion-1 emotion-2"
            type="text"
          />
        </PasswordInput__Input>
        <PasswordInput__Toggle
          as="button"
          bg="transparent"
          color="inherit"
          m={0}
          onClick={[Function]}
          p={0}
          type="button"
        >
          <button
            className="emotion-14 emotion-15"
            color="inherit"
            onClick={[Function]}
            type="button"
          >
            <Icon
              mr={2}
              name="visibilityOff"
              size={22}
              title={null}
            >
              <WithTheme(Component)
                className="emotion-8 emotion-5"
                mr={2}
                name="visibilityOff"
                size={22}
                title={null}
              >
                <Component
                  className="emotion-8 emotion-5"
                  mr={2}
                  name="visibilityOff"
                  size={22}
                  title={null}
                >
                  <StyledSvg
                    className="emotion-8 emotion-5"
                    fill="currentcolor"
                    height={22}
                    mr={2}
                    title="visibilityOff"
                    viewBox="0 0 24 24"
                    width={22}
                  >
                    <svg
                      className="emotion-5 emotion-6 emotion-7"
                      fill="currentcolor"
                      height={22}
                      title="visibilityOff"
                      viewBox="0 0 24 24"
                      width={22}
                    >
                      <path
                        d={null}
                      />
                    </svg>
                  </StyledSvg>
                </Component>
              </WithTheme(Component)>
            </Icon>
            Hide
          </button>
        </PasswordInput__Toggle>
      </div>
    </Wrapper>
  </Base>
</PasswordInput>
`;
